File: frontend/src/api/qa.ts
import api from './client'

export interface QAIssue {
  term_id: number
  source_term: string
  expected_translation: string
  found: boolean
  do_not_translate: boolean
  transliterate: boolean
}

export async function checkGlossaryConsistency(params: {
  source_text: string
  translated_text: string
  language_id: number
}): Promise<{ issues: QAIssue[] }> {
  const { data } = await api.post('/qa/glossary-check', params)
  return data
}

File: backend/app/schemas/qa.py
"""Pydantic schemas for QA endpoints."""

from pydantic import BaseModel


class GlossaryCheckRequest(BaseModel):
    source_text: str
    translated_text: str
    language_id: int


class QAIssue(BaseModel):
    term_id: int
    source_term: str
    expected_translation: str
    found: bool
    do_not_translate: bool = False
    transliterate: bool = False


class GlossaryCheckResponse(BaseModel):
    issues: list[QAIssue]
